veggieseasons/lib/main.dart

```diff
@@ -16,16 +16,76 @@ void main() {
     DeviceOrientation.portraitDown,
   ]);
 
-  runApp(
-    ScopedModel<AppState>(
-      model: AppState(),
+  // The root scope is usually injected automatically by the
+  // CupertinoApp/MaterialApp. However, in this particular app there are widgets
+  // above the CupertinoApp that need their state restored, so we are injecting
+  // the root scope manually above everything.
+  runApp(RestorationScope.root(
+    restorationId: const RestorationId('root'),
+    child: VeggieApp(),
+  ));
+}
+
+// A new stateful widget that owns the [AppState], which contains state that
+// needs to be restored.
+class VeggieApp extends StatefulWidget {
+  const VeggieApp({Key key}) : super(key: key);
+
+  @override
+  State<VeggieApp> createState() => _VeggieAppState();
+}
+
+class _VeggieAppState extends State<VeggieApp> with RestorationMixin {
+  RestorableAppState appState = RestorableAppState();
+
+  @override
+  void restoreState(bool initialRestore) {
+    registerForRestoration(appState, const RestorationId('state'));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ScopedModel<AppState>(
+      model: appState.value,
       child: ScopedModel<Preferences>(
         model: Preferences()..load(),
         child: CupertinoApp(
           debugShowCheckedModeBanner: false,
           home: HomeScreen(),
+          restorationId: const RestorationId('home'),
         ),
       ),
-    ),
-  );
+    );
+  }
+
+  @override
+  RestorationId get restorationId => const RestorationId('app');
+}
+
+// A custom RestorableProperty that saves and restores the data in an [AppState]
+// object.
+class RestorableAppState extends RestorableListenable<AppState> {
```

Comment:
is RestorableListenable a RestorableProperty? If so it would be nice if it had Property in the name for clarity.
Ditto RestorableAppState.